src/command_modules/azure-cli-backup/azure/cli/command_modules/backup/tests/latest/test_backup_commands.py

```diff
@@ -344,6 +405,9 @@ def test_backup_protection(self, resource_group, vault_name, vm_name):
         self.cmd('backup container list -v {vault} -g {rg}',
                  checks=self.check("length(@)", 0))
 
+        isenabled_check = self.cmd('backup protection isenabled-for-vm --vm-id {vm_id}').output
```

Comment:
`--vm-id` should be `--resource` and should accept the name or ID of the resource. In this case, that resource will be a VM, so there's no need to expose the other params like `resource_type`, `resource_namespace`, etc. When other types are supported in the future, you can expose these arguments.

Reply:
Per our meeting, just make this `--vm` to accept a name or ID.
https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#supporting-name-or-id-parameters